server/apps/publish/archive_publish.py

```diff
@@ -263,22 +267,23 @@ def process_takes(self, updates_of_take_to_be_published, package_id, original_of
                     break
 
             if sequence_num_of_take_to_be_published:
-                for sequence in range(sequence_num_of_take_to_be_published, 0, -1):
-                    previous_take_ref = next(ref for ref in take_refs if ref.get(SEQUENCE) == sequence)
-                    if previous_take_ref[GUID_FIELD] != take_article_id:
-                        previous_take = super().find_one(req=None, _id=previous_take_ref[GUID_FIELD])
+                if self.published_state != "killed":
+                    for sequence in range(sequence_num_of_take_to_be_published, 0, -1):
+                        previous_take_ref = next(ref for ref in take_refs if ref.get(SEQUENCE) == sequence)
```

Comment:
Can it happen that no `ref` matches the condition? If yes, then I suggest passing a default argument to `next()` in order to avoid (probably unwanted?) `StopIteration` error.